torch/distributed/_spmd/api.py

```diff
@@ -393,6 +340,7 @@ class _CompiledResult:
 def _compile(
     func: Callable,
     module_override: Optional[Dict[Type[Any], Override]],
+    parallel_mode: Optional[ParallelMode],
```

Comment:
Does this mean, we will have dedicated DDP and FSDp parallel mode which can be plugged in independently?

Reply:
Yeah, I would like to add a dedicated DataParallelMode later where it consists of 3 styles as we described in the design (default, replicate, fully_shard). then base on the style the expansion would be different and would generate its corresponding pattern. I am not sure if we want to shard the parameters/states in the `parallel_mode.expand` api directly, so looking for some suggestions :)

Reply:
Consolidating all the information into once place is good. However, this can also pose an issue that there will be combinatorial default modes we will have to set (e.g., DDP, FSDP, TP, SPARSE and they can be combined). Additionally, if users only want to override the transformation part (users may not have good knowledge or don't care about expansion), this seems to be a burden.

Reply:
Oh I think for composable parallelism (i.e. DDP/FSDP/TP/Sparse/Pipeline), we should either create a "manual" parallel mode that manually composes all different expansion and optimizations, or we should rely on the "auto" parallel mode directly? It's a hard to answer question before we really hit such a case though..